### PR TITLE
JAMES-2884 Rename jmap modules to jmap-draft

### PR DESCRIPTION
**Pull request by cketti**
First step of the plan outlined in https://www.mail-archive.com/server-dev@james.apache.org/msg62072.html

- Rename all Maven modules with `jmap` in their name to use `jmap-draft` instead
- Rename `JMAPServer` to `JMAPDraftServer`

https://issues.apache.org/jira/browse/JAMES-2884

### CODE REVIEW

File: server/data/data-jmap-draft-cassandra/pom.xml [hunk 1]
```diff
@@ -28,10 +28,10 @@
         <relativePath>../../pom.xml</relativePath>
     </parent>
 
-    <artifactId>james-server-data-jmap-cassandra</artifactId>
+    <artifactId>james-server-data-jmap-draft-cassandra</artifactId>
     <packaging>jar</packaging>
 
-    <name>Apache James :: Server :: Data :: JMAP :: Cassandra persistance</name>
+    <name>Apache James :: Server :: Data :: JMAP (draft) :: Cassandra persistence</name>
```
Comment (cketti) [inline server/data/data-jmap-draft-cassandra/pom.xml]: Note: Fixed unrelated typo here (persistance -> persistence)

Comment (chibenwa) [reply server/data/data-jmap-draft-cassandra/pom.xml]: Regarding that one, maybe we should not do the same split on persistence APIs.

My thaughts:

 - There are some that both protocols gonna use, and in that case we want the underlying data to keep the same.
 - It's not a bad thing if some APIs are used only by jmap or jmap-draft

I propose to keep it `/server/data/data-jmap-api`...

File: server/data/data-jmap-draft/pom.xml [hunk 2]
```diff
@@ -27,10 +27,10 @@
         <relativePath>../../pom.xml</relativePath>
     </parent>
 
-    <artifactId>james-server-data-jmap</artifactId>
+    <artifactId>james-server-data-jmap-draft</artifactId>
```
Comment (chibenwa) [inline server/data/data-jmap-draft/pom.xml]: We should keep it as `james-server-data-jmap` as explained above.

File: server/protocols/jmap-draft/pom.xml [hunk 3]
```diff
@@ -27,10 +27,10 @@
         <relativePath>../../pom.xml</relativePath>
```
Comment (chibenwa) [inline server/protocols/jmap-draft/pom.xml]: Please add a `.draft.` part in the package name in this module too.

File: server/container/guice/protocols/jmap-draft/src/main/java/org/apache/james/jmap/JMAPModule.java [hunk 4]
```diff
@@ -88,7 +88,7 @@ protected void configure() {
             .bind(CamelMailetContainerModule.DefaultProcessorsConfigurationSupplier.class)
```
[the remark this answers is not in the document]
Comment (cketti) [reply server/container/guice/protocols/jmap-draft/src/main/java/org/apache/james/jmap/JMAPModule.java]: Done.

File: server/container/guice/protocols/jmap-draft/src/main/java/org/apache/james/utils/JmapGuiceProbe.java [hunk 5]
```diff
@@ -40,13 +40,13 @@
 public class JmapGuiceProbe implements GuiceProbe {
```
[the remark this answers is not in the document]
Comment (cketti) [reply server/container/guice/protocols/jmap-draft/src/main/java/org/apache/james/utils/JmapGuiceProbe.java]: Done. I went with changing the package name.

File: server/protocols/jmap-draft/pom.xml (hunk 3 above)
Comment (cketti) [reply server/protocols/jmap-draft/pom.xml]: Done.

**Comment by cketti** [comment]
I had another run at this. 

- The data modules are now left untouched.
- Most of the code was moved to `org.apache.james.jmap.draft`
- I left the Mailets under `org.apache.james.jmap.mailet` because I believe the fully qualified name could be used in configuration files. I'd be happy to move them, too, if that assumption is wrong.

**Comment by chibenwa** [comment]
`I left the Mailets under org.apache.james.jmap.mailet because I believe the fully qualified name could be used in configuration files. I'd be happy to move them, too, if that assumption is wrong.`

That assumption is right (cf mailetContainer.xml)

My personal believe regarding mailets is that new & old spec can share the same mailet (underlying logic did not change). Thus extracting JMAP mailet under a `server/mailet/jmap` maven module would make sens to me. IMO let's do that when we support Vacations on top of `JMAP`

**Comment by chibenwa** [comment]
Hello @cketti ,

There are some failed tests after your rename proposal, I made some corrections to your work here, feel free to cherry-pick my commits here!

Issues:
 - MailboxListeners are also referenced by FQCN (delivery groups in the MailboxListener workQueue)
 - Cucumber tests where referring FQCN for some "glue"

Best regards,

Benoit

**Comment by cketti** [comment]
@chibenwa Thanks :+1:

Is there an easy way to build the project, then start it up and run the integration tests using Docker?

**Comment by chibenwa** [comment]
> Is there an easy way to build the project, then start it up and run the integration tests using Docker?

The integration tests are fully integrated to the junit test suite. We rely on testcontainer leveraging docker-java in order to start dependencies then run James directly in the testing JVM

For running them, having docker working and writting `mvn test` will atually play them.

If you are looking for a CI have a look at https://github.com/linagora/james-jenkins

Also, we have a CI running, but on `linagora/james-project`

**Comment by chibenwa** [comment]
Hi @cketti ,

I just merged this work, can you close this PR?

**Pull request closed without merging**